### ISSUE TITLE
[tune] Add support for histogram metric reporting

### ISSUE THREAD

**Issue by roireshef** [ray-project/ray#4953]
Currently only scalar metrics are supported when using _custom_metrics_. It is going to be useful to have Tensorboard's histograms as well, for example to plot action distribution in a training iteration, or episode.

For that, tensorboardX's [make_histogram method](https://github.com/lanpa/tensorboardX/blob/master/tensorboardX/summary.py#L146) can used. 

Nevertheless, some special treatment should be added to allow that, since _custom_metrics_ assumes scalar metrics only, and therefor [_summarize_episodes_ computes different aggregations on those scalars](https://github.com/ray-project/ray/blob/master/python/ray/rllib/evaluation/metrics.py#L124), which aren't suitable for histogram data.

**Comment by RoiGM** [comment]
@ericl I Can help implementing this if we agree on the design.

**Comment by ericl** [comment]
@RoiGM histogram would be nice to have. How would you specify the histogram type aggregation here?

Probably tune also needs to be changed to support logging histogram outputs.

**Comment by richardliaw** [comment]
For histograms and images, what are thoughts on something like:

```python

def trainable(config):
    for i in range(50):
		results = {
           "training_acc": 50,
		   "distribution": tune.result.histogram(nparray, max_bins=5),
		   "distribution_raw": tune.result.histogram_raw(min, max, num, sum, sum_squares, bucket_limits, bucket_counts, global_step=None, walltime=None),
		   "image": tune.result.image(nparray) # Assume NHWC
	    }
	    tune.track.log(results)

tune.run(trainable)
```

**Comment by roireshef** [comment]
Hi @richardliaw, sorry for not getting back to you in a while. I finally have some time to share for the benefit of this awesome project. I have implemented histogram reporting from RLlib -> TB. Although a a bit different than you suggested, I found it to be generic and short. LMK what you think, I find this feature super-powerful for debugging training sessions:

https://github.com/ray-project/ray/pull/6942
